scarlet2/initialization.py

```diff
@@ -0,0 +1,331 @@
+        if (
+            cols > 2
+        ):  # Check if there are more than 2 columns to avoid duplicate corner values
+            perimeter_values.extend(array_2d[i][-1] for i in range(1, rows - 1))
+
+    return perimeter_values
+
+
+# create a 2D Gaussian for simple feature free Initializations
+def gaussian_2d(x, y, mu_x, mu_y, sigma_x, sigma_y):
+    """2D Gaussian function.
+    Parameters
+    ----------
+    x : jnp.ndarray
+    y : jnp.ndarray
+    mu_x : float mean of the Gaussian in the x direction
+    mu_y : float mean of the Gaussian in the y direction
+    sigma_x : float standard deviation of the Gaussian in the x direction
+    sigma_y : float standard deviation of the Gaussian in the y direction
+
+    Returns
+    -------
+    jnp.ndarray 2D Gaussian profile
+    """
+    return jnp.exp(
+        -((x - mu_x) ** 2 / (2 * sigma_x**2) + (y - mu_y) ** 2 / (2 * sigma_y**2))
+    )
+
+
+@partial(jax.jit, static_argnums=0)
+def create_gaussian_array(boxsize, sigma_x=1, sigma_y=1, theta=0):
+    """Create a 2D array with a Gaussian profile in the center.
+    Parameters
+    ----------
+    boxsize : int size of the bounding box around source
+    sigma_x : float standard deviation of the Gaussian in the x direction
+    sigma_y : float standard deviation of the Gaussian in the y direction
+    theta : float rotation angle of the 2D Gaussian
+
+    Returns
+    -------
+    jnp.ndarray 2D array with a Gaussian profile rotated by theta
+    """
+    x = jnp.linspace(-boxsize / 2, boxsize / 2, boxsize)
+    x, y = jnp.meshgrid(x, x)
+    # Center of the Gaussian profile
+    mu_x, mu_y = 0, 0
+    # Generate the Gaussian profile
+    gaussian_profile = gaussian_2d(x, y, mu_x, mu_y, sigma_x, sigma_y)
+    gaussian_profile = (gaussian_profile - jnp.min(gaussian_profile)) / (
+        jnp.max(gaussian_profile) - jnp.min(gaussian_profile)
+    )
+    gaussian_profile = jnp.array(gaussian_profile, dtype=jnp.float32)
+    gaussian_profile = jnp.expand_dims(gaussian_profile, axis=-1)
+    data = dm_pix.rotate(gaussian_profile, theta)
+    return jnp.squeeze(data)
+
+
+def fit_morph_params(data, center, bx):
+    """Fit a 2D Gaussian to the source.
+
+    Parameters
+    ----------
+    data : jnp.ndarray 2D cutout of the source in single band
+    center : tuple (int, int) central pixel of the source
+    bx : int (boxsize)
+
+    Returns
+    -------
+    best_params : jnp.ndarray best fit parameters for the 2D Gaussian
+
+    """
+    data_cut = (data - jnp.min(data)) / (jnp.max(data) - jnp.min(data))
+    central_gaussian = create_gaussian_array(bx, 1.75, 1.75, 0)
+    fn = lambda x: jnp.mean(
+        central_gaussian * (data_cut - create_gaussian_array(bx, x[0], x[1], x[2])) ** 2
+    )
+    x0 = jnp.array([1.5, 1.5, 0])
+    lbfgsb = ScipyBoundedMinimize(fun=fn, method="l-bfgs-b", jit=True)
+    lower_bounds = jnp.array([1, 1, jnp.pi / 2])
+    upper_bounds = jnp.array([7, 7, jnp.pi])
+    bounds = (lower_bounds, upper_bounds)
```

Comment:
these initial values I can understand but the limits are entirely arbitrary. They may well not work in practice. It would be much better to replace this by a unconstrained optimization on log(s)

Reply:
Removed fitting routine